app/src/main/java/org/schabi/newpipe/settings/SettingsActivity.java

```diff
@@ -91,14 +171,211 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     @Override
     public boolean onPreferenceStartFragment(final PreferenceFragmentCompat caller,
                                              final Preference preference) {
-        final Fragment fragment = Fragment
-                .instantiate(this, preference.getFragment(), preference.getExtras());
+        showSettingsFragment(instantiateFragment(preference.getFragment()));
+        return true;
+    }
+
+    private Fragment instantiateFragment(@NonNull final String className) {
+        return getSupportFragmentManager()
+                .getFragmentFactory()
+                .instantiate(this.getClassLoader(), className);
+    }
+
+    private void showSettingsFragment(final Fragment fragment) {
         getSupportFragmentManager().beginTransaction()
                 .setCustomAnimations(R.animator.custom_fade_in, R.animator.custom_fade_out,
                         R.animator.custom_fade_in, R.animator.custom_fade_out)
-                .replace(R.id.settings_fragment_holder, fragment)
+                .replace(FRAGMENT_HOLDER_ID, fragment)
                 .addToBackStack(null)
                 .commit();
-        return true;
     }
+
+    @Override
+    protected void onDestroy() {
+        setMenuSearchItem(null);
+        searchFragment = null;
+        super.onDestroy();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Search
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Search
+
+    private void initSearch(
+            final SettingsLayoutBinding settingsLayoutBinding,
+            final boolean restored
+    ) {
+        searchContainer =
+                settingsLayoutBinding.settingsToolbarLayout.toolbar
+                        .findViewById(R.id.toolbar_search_container);
+
+        // Configure input field for search
+        searchEditText = searchContainer.findViewById(R.id.toolbar_search_edit_text);
+        RxTextView.textChanges(searchEditText)
+                // Wait some time after the last input before actually searching
+                .debounce(200, TimeUnit.MILLISECONDS)
+                .subscribe(v -> runOnUiThread(this::onSearchChanged));
+
+        // Configure clear button
+        searchContainer.findViewById(R.id.toolbar_search_clear)
+                .setOnClickListener(ev -> resetSearchText());
+
+        prepareSearchConfig();
```

Comment:
I'd rename this to something else, `prepareSearchConfig` is misleading imho since you don't actually do anything with `config` below.